fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateRoutineLoadInfo.java

```diff
@@ -0,0 +1,525 @@
+                    columnSeparator = new Separator(separator, oriSeparator);
+                } else if (loadProperty instanceof LoadColumnClause) {
+                    if (isMultiTable) {
+                        throw new AnalysisException("Multi-table load does not support setting columns info");
+                    }
+                    List<ImportColumnDesc> importColumnDescList = new ArrayList<>();
+                    for (LoadColumnDesc columnDesc : ((LoadColumnClause) loadProperty).getColumns()) {
+                        if (columnDesc.getExpression() != null) {
+                            Expr expr = translateToLegacyExpr(columnDesc.getExpression(), analyzer,
+                                    context, cascadesContext);
+                            importColumnDescList.add(new ImportColumnDesc(columnDesc.getColumnName(), expr));
+                        } else {
+                            importColumnDescList.add(new ImportColumnDesc(columnDesc.getColumnName(), null));
+                        }
+                    }
+                    importColumnsStmt = new ImportColumnsStmt(importColumnDescList);
+                } else if (loadProperty instanceof LoadWhereClause) {
+                    if (isMultiTable) {
+                        throw new AnalysisException("Multi-table load does not support setting columns info");
+                    }
+                    Expr expr = translateToLegacyExpr(((LoadWhereClause) loadProperty).getExpression(),
+                            analyzer, context, cascadesContext);
+                    if (((LoadWhereClause) loadProperty).isPreceding()) {
+                        precedingImportWhereStmt = new ImportWhereStmt(expr,
+                                ((LoadWhereClause) loadProperty).isPreceding());
+                    } else {
+                        importWhereStmt = new ImportWhereStmt(expr, ((LoadWhereClause) loadProperty).isPreceding());
+                    }
+                } else if (loadProperty instanceof LoadPartitionNames) {
+                    partitionNames = new PartitionNames(((LoadPartitionNames) loadProperty).isTemp(),
+                            ((LoadPartitionNames) loadProperty).getPartitionNames());
+                } else if (loadProperty instanceof LoadDeleteOnClause) {
+                    Expr expr = translateToLegacyExpr(((LoadDeleteOnClause) loadProperty).getExpression(),
+                            analyzer, context, cascadesContext);
+                    importDeleteOnStmt = new ImportDeleteOnStmt(expr);
+                } else if (loadProperty instanceof LoadSequenceClause) {
+                    importSequenceStmt = new ImportSequenceStmt(
+                            ((LoadSequenceClause) loadProperty).getSequenceColName());
+                }
+            }
+        }
+        routineLoadDesc = new RoutineLoadDesc(columnSeparator, lineDelimiter, importColumnsStmt,
```

Comment:
Better to create a new RoutineLoadInfo class and remove the dependence of ImportColumnsStmt, ImportWhereStmt. Just keep Expression as its member variable. And add setRoutineLoadDesc(RoutineLoadInfo info) into RoutineLoadJob to use the new class.(you can do it in next pr)

Reply:
got it